cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperatorTest.java

```diff
@@ -0,0 +1,120 @@
+import io.strimzi.api.kafka.model.KafkaClusterRebalanceSpecBuilder;
+import io.strimzi.api.kafka.model.status.Condition;
+import io.strimzi.operator.KubernetesVersion;
+import io.strimzi.operator.PlatformFeaturesAvailability;
+import io.strimzi.operator.cluster.ResourceUtils;
+import io.strimzi.operator.cluster.model.CruiseControl;
+import io.strimzi.operator.cluster.operator.assembly.cruisecontrol.MockCruiseControl;
+import io.strimzi.operator.cluster.operator.resource.ResourceOperatorSupplier;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.operator.resource.CrdOperator;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockserver.integration.ClientAndServer;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(VertxExtension.class)
+public class KafkaClusterRebalanceAssemblyOperatorTest {
+
+    private static final String HOST = "localhost";
+    private static final String CLUSTER_NAMESPACE = "cruise-control-namespace";
+    private static final String CLUSTER_NAME = "cruise-control-test-cluster";
+
+    private final KubernetesVersion kubernetesVersion = KubernetesVersion.V1_11;
+
+    private static final Logger log = LogManager.getLogger(KafkaClusterRebalanceAssemblyOperatorTest.class.getName());
+
+    private static ClientAndServer ccServer;
+
+    @BeforeAll
+    public static void before() throws IOException, URISyntaxException {
+        ccServer = MockCruiseControl.getCCServer(CruiseControl.REST_API_PORT, 0);
+    }
+
+    @AfterAll
+    public static void after() {
+        ccServer.stop();
+    }
+
+    @Test
+    public void testNewRebalance(Vertx vertx, VertxTestContext context) {
+
+        KafkaClusterRebalanceBuilder kcrBuilder = new KafkaClusterRebalanceBuilder();
+
+        Map<String, String> labels = new HashMap<>();
+        labels.put(Labels.STRIMZI_CLUSTER_LABEL, "my-test-cluster");
+        ObjectMeta meta = new ObjectMetaBuilder().withLabels(labels).withName(CLUSTER_NAME).withNamespace(CLUSTER_NAMESPACE).build();
+
+        KafkaClusterRebalanceSpec rebalanceSpec = new KafkaClusterRebalanceSpecBuilder().build();
+
+        Condition newRebalanceCondition = new Condition();
+        newRebalanceCondition.setType(String.valueOf(KafkaClusterRebalanceAssemblyOperator.State.New));
```

Comment:
a new `KafkaClusterRebalance` resource doesn't need to have an initial condition with status `New`. Of course, it's something we cannot ask to the user. For this reason when the reconcile starts, it's the operator to set this status to `New` when the condition is null (https://github.com/kyguy/strimzi-kafka-operator/blob/kafka-cluster-rebalance-operator/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java#L605)